pkg/client/preflight.go

```diff
@@ -41,40 +63,42 @@ func (c *SonobuoyClient) PreflightChecks(cfg *PreflightConfig) []error {
 	}
 
 	errors := []error{}
-
 	for _, check := range preflightChecks {
 		if err := check(client, cfg); err != nil {
 			errors = append(errors, err)
 		}
 	}
+
 	return errors
 }
 
-const (
-	kubeSystemNamespace = "kube-system"
-	kubeDNSLabelKey     = "k8s-app"
-	kubeDNSLabelValue   = "kube-dns"
-	coreDNSLabelValue   = "coredns"
-)
-
 func preflightDNSCheck(client kubernetes.Interface, cfg *PreflightConfig) error {
-	var dnsLabels = []string{
-		kubeDNSLabelValue,
-		coreDNSLabelValue,
+	return dnsCheck(
+		client.CoreV1().Pods(kubeSystemNamespace).List,
```

Comment:
I think this can be simplified to just run `dnsCheck`.  Passing the dnsLabels & pod list command doesn't add much and just complicates.

Reply:
It allows us (for the first time) to run tests against those functions. 100% of our logic requires a single method (e.g. list pods or whichever method you're talking about) but our method signatures require the entire kubernetes.Interface which makes them effectively untestable.

Then the dnsLabels stuff _could_ be hardcoded into the method but it's just a preference that things like that be injected as well. It also makes it more clear when looking at the method signature what you need to do/what it may be looking for; otherwise, you need to look up which constants actually get used in the method. Again, I think of this part of the code a little more habit/preference though.